samples/caf_sensor_manager/src/modules/workload_sim.c

```diff
@@ -88,15 +88,12 @@ static bool event_handler(const struct app_event_header *aeh)
 		return false;
 	}
 
-	if (is_module_state_event(aeh)) {
-		const struct module_state_event *event = cast_module_state_event(aeh);
+	if (is_config_event(aeh)) {
+		
+		const struct config_event *event = cast_config_event(aeh);
 
-		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
-			if (!init()) {
-				module_set_state(MODULE_STATE_READY);
-			} else {
-				module_set_state(MODULE_STATE_ERROR);
-			}
+		if(event->is_ready == true) {
```

Comment:
space is missing
do not compare to true (true is ambiguous), if needed compare to false (i.e. `a != false` or `a == false`). Better avoid comparing at all (i.e. `if (a)` or `if (!a)`)